griptape/tools/openweather_client/tool.py

```diff
@@ -0,0 +1,48 @@
+from __future__ import annotations
+from griptape.artifacts import TextArtifact, ErrorArtifact
+from griptape.tools import BaseTool
+from griptape.utils.decorators import activity
+from schema import Schema, Literal
+from attr import define, field
+from typing import Optional
+import requests
+import logging
+
+
+@define
+class OpenWeatherClient(BaseTool):
+    BASE_URL = "http://api.openweathermap.org/data/2.5/weather"
+    api_key: str = field(kw_only=True)
+    units: Optional[str] = field(default="imperial", kw_only=True)
+
+    @activity(
+        config={
+            "description": "Can be used to fetch weather data for a given city using the OpenWeather API. Temperatures are returned in {self.units} by default.",  # Modified description
```

Comment:
This won't work because `description` is a Jinja template that gets generated dynamically in `ActivityMixin.activity_description`. As a workaround we inject Jinja variables into `schema_template_args`. Check out how [we do it](https://github.com/griptape-ai/griptape/blob/main/griptape/tools/vector_store_client/tool.py#L20) in `VectorStoreClient`.